### PR TITLE
Meta: Extend `rgh-linkify-features` to more places

### PR DESCRIPTION
**Pull request by kidonng**
## Test URLs

- [`isCommitList`](https://github.com/refined-github/refined-github/commits/6d5bda3b9db8b8d1514af40ff53628f198f4478d)
- [`isSingleCommit`](https://github.com/refined-github/refined-github/commit/6d5bda3b9db8b8d1514af40ff53628f198f4478d)
- [`isRepoTree`](https://github.com/refined-github/refined-github/tree/6d5bda3b9db8b8d1514af40ff53628f198f4478d) (one gotcha: it won't work for links loaded asynchronously, but we probably don't care)

## Screenshot

Before:
Screen Shot 2021-11-01 at 23 27 01
Screen Shot 2021-11-01 at 23 26 57
Screen Shot 2021-11-01 at 23 44 00

After:
Screen Shot 2021-11-01 at 23 26 43
Screen Shot 2021-11-01 at 23 26 48
Screen Shot 2021-11-01 at 23 40 28

### CODE REVIEW

File: source/github-helpers/index.ts
Comment (kidonng) [inline source/github-helpers/index.ts:116]: I'm not sure if the trailing slash is intentional, but it _is_ a bug so I fixed it 🤷

File: source/features/rgh-linkify-features.tsx
Comment (kidonng) [inline source/features/rgh-linkify-features.tsx:19]: If the blueish looks too much, we can use `.Link--primary` instead which only highlights in blue when hovering.

File: source/features/rgh-linkify-features.tsx
Comment (kidonng) [inline source/features/rgh-linkify-features.tsx:33]: About the new convoluted selectors: https://github.com/refined-github/refined-github/issues/4135#issuecomment-830171333

File: source/github-helpers/index.ts
Comment (cheap-glitch) [reply source/github-helpers/index.ts:116]: Since we have `asLongAs` now, this function doesn't have to be "negated" btw.

Comment (yakov116) [reply source/github-helpers/index.ts:116]: Can you change it to a positive function since we dont have to use exclude we can use the new option we added (That I dont remember the name)

File: source/features/rgh-linkify-features.tsx
Comment (kidonng) [inline source/features/rgh-linkify-features.tsx:44]: I didn't bother to add pages like `isCompare` but suggestions welcome.

File: source/features/rgh-linkify-features.tsx
Comment (fregante) [reply source/features/rgh-linkify-features.tsx:39]: Also one selector per line would be good

File: source/features/rgh-linkify-features.tsx
Comment (fregante) [inline source/features/rgh-linkify-features.tsx:14]: Do we have a helper function for this? It's also generated in the options page.

**Comment by kidonng** [comment]
Woops, [a side effect](https://togithub.com/refined-github/refined-github/issues/4542#event-5320805189) of the new selectors::

Screen Shot 2021-11-02 at 00 19 16

But I kinda like it

File: source/features/rgh-linkify-features.tsx
Comment (fregante) [inline source/features/rgh-linkify-features.tsx:17]: I'm concerned about this logic. How are we sure that the possibleLink isn't the whole line instead of just the feature name?

I'd like to extract this part into a function that accepts a `code` element and linkifies it the best way it can.

Comment (kidonng) [reply source/features/rgh-linkify-features.tsx:17]: Good call. I'll add a extra check

File: source/helpers/feature-link.ts
Comment (yakov116) [inline source/helpers/feature-link.ts:1]: I would put this in the index don't think it warrants its own file. JMO

@refined-github/maintainers?

Comment (kidonng) [reply source/helpers/feature-link.ts:1]: We have other one-liners like `browser-detection.ts`. I don't think it's a big deal.

What index are you referring to? AFAICT `options.tsx` only imports utils from `helpers` folder.

File: source/features/rgh-linkify-features.tsx
Comment (kidonng) [reply source/features/rgh-linkify-features.tsx]: I added one

**Comment by fregante** [comment]
For isRepoTree we have an additional listener we can use, if desired

File: source/features/rgh-linkify-features.tsx
Comment (fregante) [reply source/features/rgh-linkify-features.tsx]: Let's make one https://github.com/refined-github/refined-github/blob/6d5bda3b9db8b8d1514af40ff53628f198f4478d/source/options.tsx#L116

File: source/features/rgh-linkify-features.tsx
Comment (fregante) [reply source/features/rgh-linkify-features.tsx:18]: The code would still benefit from some clarification. The variable name is codeElement but now it seems that it can also be a link. 

Can you add a comment before each `if` explaining the expected DOM?

**Comment by fregante** [comment]
If I don't review for a while feel free to merge after others’ reviews

File: source/features/rgh-feature-descriptions.tsx
Comment (yakov116) [inline source/features/rgh-feature-descriptions.tsx]: lint

Comment (kidonng) [reply source/features/rgh-feature-descriptions.tsx]: What's the issue?

File: source/options-storage.ts
Comment (kidonng) [inline source/options-storage.ts:46]: This is changed to reflect that `newFeatureName` may actually not be a feature name.

**Review by fregante — APPROVED** [review]
LGTM after this change. 

Also, slightly unrelated, I kinda wish there was a way to open the TSX file of a feature when visiting its CSS file. We could add a button somewhere.

```
if filebasename in url is included in featureList
    add link to tsx file

File: source/features/rgh-feature-descriptions.tsx
Comment (kidonng) [reply source/features/rgh-feature-descriptions.tsx:65]: There is an early return in `init` for this purpose. I would prefer readability over less duplicate code here.

**Comment by kidonng** [comment]
> Also, slightly unrelated, I kinda wish there was a way to open the TSX file of a feature when visiting its CSS file. We could add a button somewhere.
> 
> ```
> if filebasename in url is included in featureList
>     add link to tsx file
> ```
> 
> It could be part of rgh-feature-descriptions since it works on the same files and is just a button. Maybe different PR

In another PR I suppose, maybe also throw https://github.com/refined-github/refined-github/pull/5011#issuecomment-955240247 in. This PR is becoming huge despite most being refactor changes.

**Comment by kidonng** [comment]
> Also, slightly unrelated, I kinda wish there was a way to open the TSX file of a feature when visiting its CSS file. We could add a button somewhere.
> 
> ```
> if filebasename in url is included in featureList
>     add link to tsx file
> ```
> 
> It could be part of rgh-feature-descriptions since it works on the same files and is just a button. Maybe different PR

In another PR I suppose, maybe also throw https://github.com/refined-github/refined-github/pull/5011#issuecomment-955240247 in. This PR is becoming huge despite most being refactor changes.

**Review by SMaldhaheri — APPROVED** [review]

**Pull request merged**
